apps/sms/views/shared/test/unit/contacts_test.js

```diff
@@ -150,111 +38,82 @@ suite('Contacts', function(done) {
     window.fb = mockReaderUtils;
   });
 
-  teardown(function() {
-    navigator.mozContacts.mHistory.length = 0;
-  });
-
   suiteTeardown(function() {
     navigator.mozContacts = nativeMozContacts;
     window.fb = realFb;
+    //Restoring the native API
+    navigator.mozContacts.find.restore();
   });
 
   suite('Contacts.findContactByString, single-term', function() {
 
     test('(string[tel,givenName,familyName], ...) Match', function(done) {
-      var mozContacts = navigator.mozContacts;
+      var result = MockContact.list();
 
-      Contacts.findContactByString('O\'Hare').then(function(contacts) {
-        var mHistory = mozContacts.mHistory;
+      navigator.mozContacts.find.returns(Promise.resolve(result));
```

Comment:
The best would be to have a `withArgs`; because you'll need to use it everywhere in the file and to make it more readable, you can use a function:
```javascript
function withFilter(str) {
  return sinon.match({ filterValue: str });
}

navigator.mozContacts.find.withArgs(
  withFilter('O\'Hare' )
).returns(
  Promise.resolve(result)
);
```

What do you think ?